Resize disk reconfigure screen

evertmulder opened the pull request:
PR to support resize disk from reconfigure screen.

See: https://github.com/ManageIQ/manageiq-ui-classic/issues/3127
![screen shot 2018-01-12 at 15 03 47](https://user-images.githubusercontent.com/61044/34879157-9f938ab4-f7ac-11e7-9152-7d0f264f4fa3.png)
![screen shot 2018-01-12 at 15 06 26](https://user-images.githubusercontent.com/61044/34879159-a2260e50-f7ac-11e7-8127-b42419a92158.png)
![screen shot 2018-01-12 at 15 09 55](https://user-images.githubusercontent.com/61044/34879165-a61271e8-f7ac-11e7-8b61-15f842e62afb.png)
![screen shot 2018-01-12 at 15 13 38](https://user-images.githubusercontent.com/61044/34879167-a8f81336-f7ac-11e7-866d-5a45fb7bbd97.png)
![screen shot 2018-01-12 at 15 15 27](https://user-images.githubusercontent.com/61044/34879172-ac6e45bc-f7ac-11e7-989a-828f28dcc6fa.png)

martinpovolny commented:
Restarted travis.

gmcculloug commented:
@dclarizio See screenshot in PR https://github.com/ManageIQ/manageiq-ui-classic/issues/3127.  Please assign these two PRs.

dclarizio commented:
@evertmulder I'm not seeing any tests, perhaps @AparnaKarve or @himdel can recommend some.

AparnaKarve commented:
@evertmulder Can you add some screenshots related to this feature?

himdel commented:
(pending core because https://github.com/ManageIQ/manageiq-providers-vmware/pull/164)

himdel commented:
@evertmulder You'll need to rebase - no merge commits in a PR please ;)

(This should also fix the ruby specs.)

JS specs failure looks related

himdel commented on app/assets/javascripts/controllers/reconfigure/reconfigure_form_controller.js:
```diff
@@ -44,6 +46,20 @@ ManageIQ.angular.app.controller('reconfigureFormController', ['$http', '$scope',
+  vm.setEnableAddDiskButton = function () {
+    var nrDisksAfterReconfigure=0;
+    angular.forEach(vm.reconfigureModel.vmdisks, function(disk){
+        switch (disk.add_remove) {
```
Wrong indentation ;)

Please see the rest of the code climate warnings - feel free to ignore `Function updateDisksAddRemove has 36 lines of code (exceeds 25 allowed). Consider refactoring. ` but the rest is definitely relevant :).

himdel commented on app/views/vm_common/_reconfigure.html.haml:
```diff
@@ -206,10 +215,33 @@
+                                  "placeholder" => "Enter Size",
+                                  "ng-change"   => "vm.hdChanged()",
+                                  "validate-multiple" => "1",
+                                  :miqmin             => "{{ (disk.orgHdUnit=='GB')  && (1024*disk.orgHdSize+1) || (1*disk.orgHdSize+1)}}",
```
CC won't warn here, but please keep the same rules...

`:miqmin             => "{{ (disk.orgHdUnit === 'GB')  && (1024 * disk.orgHdSize + 1) || (1 * disk.orgHdSize + 1) }}",`

evertmulder replied:
@himdel Sorry, over looked this one. Should be fix now.

himdel commented:
As for tests, I'm not sure what to suggest, sorry - can't really tell the refactoring bits from the actual changes (this being one big commit).

@evertmulder Care to elaborate what this actually changes please? :)

evertmulder commented:
@himdel Thanks for all the reviews. Tomorrow I will make some time addressing the comments. Sorry for the late response, the comments are not going to my corporate mail, so I just saw them.

evertmulder commented:
I am afraid I did merge instead of rebase. Can I do something to fix this?

AparnaKarve commented:
Try removing those 2 merge commits 93bfb6599d25651f67c9d8f7cdd240472e0e66d0 and ff0f04881fa521400b53abc1f91e093698888a20 and then a rebase

evertmulder commented:
To remove the merge commit reset to my first commit and applied the codeclimate fixes on top of that.
```  
git reset --hard 47721a6aa35b0f56978fb7e43d03ecf5bdd9be74  
git push origin HEAD --force
```
I also tried to rebase the current changes in my branch (git rebase master) but ended up with all the commit in my branch. Is there a way I can apply the current changes in master on my branch without ending op with all the commits im my PR?

agrare commented:
https://github.com/ManageIQ/manageiq-providers-vmware/pull/164 has been merged

agrare commented:
@miq-bot remove-label pending core

martinpovolny commented:
Can you, please, address the @miq-bot issue?

himdel commented:
JS failures still relevant,

also, please run eslint manually on your new code, seems like CC is not reporting properly... I'm seeing at least these in a few places..

```
  error    Expected indentation of 2 spaces but found 4  indent
  error    Expected space(s) after "if"                  keyword-spacing
  warning  Expected { after 'if' condition               curly
  warning  Expected { after 'else'                       curly
```

@miq-bot add_label wip

evertmulder commented:
@himdel Fixed all eslint issues I could find in the file. Fixed 127 issues.

himdel commented:
> It seems all PRs fail with the JS testsuite

@evertmulder Sorry about that , should be fixed now, restarting travis.

That said, can you also look at https://github.com/ManageIQ/manageiq-ui-classic/pull/3128#pullrequestreview-88163708 please? :)

evertmulder commented:
@himdel Is there anything I can improve? I think I fixed all review comments ...

himdel commented:
@evertmulder Sorry about the delay, there was a spec failure but it was on our side..
Restarted, looking again.

simaishi commented:
@himdel can this be `gaprindashvili/yes`? We need reconfigure disk feature in G-branch (https://github.com/ManageIQ/manageiq/pull/16711#issuecomment-393885855)

himdel commented:
@simaishi Assuming https://github.com/ManageIQ/manageiq-providers-vmware/pull/164 can be, I think there is no problem here.

simaishi commented:
Gaprindashvili backport details:

```
$ git log -1
commit c1d8f212ec734e1dc4f9ede5358d12b7128f3274
Author: Martin Hradil <himdel@seznam.cz>
Date:   Wed Feb 14 13:35:18 2018 +0000

    Merge pull request #3128 from evertmulder/ResizeDisk_ReconfigureScreen
    
    Resize disk reconfigure screen
    (cherry picked from commit 8835899188969665caca9894400db51c48dbd645)
```

simaishi commented:
Thanks Martin, backported VMware PR as well.
